plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -1576,6 +1584,47 @@ private ColumnMapping uuidColumnMapping()
                 uuidWriteFunction());
     }
 
+    private static ColumnMapping vectorColumnMapping(ConnectorSession session)
+    {
+        return ColumnMapping.objectMapping(
+                new ArrayType(REAL),
+                vectorReadFunction(),
+                vectorWriteFunction(session),
+                DISABLE_PUSHDOWN);
+    }
+
+    private static ObjectReadFunction vectorReadFunction()
+    {
+        return ObjectReadFunction.of(Block.class, (resultSet, columnIndex) -> {
+            // getArray is unsupported for vectors type
+            String result = resultSet.getString(columnIndex);
+            BlockBuilder builder = REAL.createBlockBuilder(null, 10);
+            if (result == null) {
+                builder.appendNull();
+            }
+            else {
+                String[] vectors = result.substring(1, result.length() - 1).split(",");
+                for (String vector : vectors) {
+                    requireNonNull(vector, "vector is null");
```

Comment:
Is the `requireNonNull` necessary ? It seems that `.spilt` should never return a null element in the result.